File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        Queue<Integer> queue = new ArrayDeque<>();
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for (int priority : priorities) {
            queue.offer(priority);
            pq.offer(priority);
        }
        
        int answer = 0;
        int max_p = pq.poll();
        while (!queue.isEmpty()) {
            int p = queue.poll();
            if (max_p == p) {
                answer++;
                if (location == 0)
                    return answer;
                max_p = pq.poll();
                location--;
            }
            else {
                queue.offer(p);
                if (location == 0)
                    location = queue.size() - 1;
                else
                    location--;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    
    while priorities:
        now_max = max(priorities)
        if priorities[0] < now_max:
            priorities.append(priorities.pop(0))
            if location == 0:
                location = len(priorities) - 1
            else:
                location -= 1
        else:
            answer += 1
            if location == 0:
                return answer
            else:
                location -= 1
            priorities.pop(0)
    
    return answer